scripts/release/build-commands/update-package-versions.js

```diff
@@ -30,30 +30,49 @@ const update = async ({cwd, dry, packages, version}) => {
     const updateProjectPackage = async project => {
       const path = join(cwd, 'packages', project, 'package.json');
       const json = await readJson(path);
+      const prerelease = semver.prerelease(version);
 
       // Unstable packages (eg version < 1.0) are treated specially:
       // Rather than use the release version (eg 16.1.0)-
       // We just auto-increment the minor version (eg 0.1.0 -> 0.2.0).
       // If we're doing a prerelease, we also append the suffix (eg 0.2.0-beta).
       if (semver.lt(json.version, '1.0.0')) {
-        const prerelease = semver.prerelease(version);
         let suffix = '';
         if (prerelease) {
           suffix = `-${prerelease.join('.')}`;
         }
 
-        json.version = `0.${semver.minor(json.version) + 1}.0${suffix}`;
+        // If this is a new pre-release, increment the minor.
+        // Else just increment (or remove) the pre-release suffix.
+        // This way our minor version isn't incremented unnecessarily with each prerelease.
+        const minor = semver.prerelease(json.version)
+          ? semver.minor(json.version)
+          : semver.minor(json.version) + 1;
+
+        json.version = `0.${minor}.0${suffix}`;
       } else {
         json.version = version;
       }
 
       if (project !== 'react') {
-        const peerVersion = json.peerDependencies.react.replace('^', '');
+        let peerVersion = json.peerDependencies.react.replace('^', '');
+        if (peerVersion.includes(' || ')) {
+          peerVersion = peerVersion.split(' || ')[0];
```

Comment:
This assumes no one manually edits them (e.g. to reorder). It's probably a fine assumption to make but would be nice to leave a comment with expected format. Or assert it.